contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
-            const erc721AddAuthorizedAddressSubmitLog = erc721AddAuthorizedAddressSubmitRes
-                .logs[0] as LogWithDecodedArgs<AssetProxyOwnerSubmissionEventArgs>;
-
-            const registerAssetProxyTxId = registerAssetProxySubmitLog.args.transactionId;
-            const erc20AddAuthorizedAddressTxId = erc20AddAuthorizedAddressSubmitLog.args.transactionId;
-            const erc721AddAuthorizedAddressTxId = erc721AddAuthorizedAddressSubmitLog.args.transactionId;
-
-            await assetProxyOwnerWrapper.confirmTransactionAsync(registerAssetProxyTxId, owners[1]);
-            await assetProxyOwnerWrapper.confirmTransactionAsync(erc20AddAuthorizedAddressTxId, owners[1]);
-            await assetProxyOwnerWrapper.confirmTransactionAsync(erc721AddAuthorizedAddressTxId, owners[1]);
-            await increaseTimeAndMineBlockAsync(SECONDS_TIME_LOCKED.toNumber());
-            await assetProxyOwnerWrapper.executeTransactionAsync(registerAssetProxyTxId, owners[0]);
-            await assetProxyOwnerWrapper.executeTransactionAsync(erc20AddAuthorizedAddressTxId, owners[0], {
-                gas: constants.MAX_EXECUTE_TRANSACTION_GAS,
+    blockchainTests.resets('executeTransaction', () => {
+        function assertReceiverCalledFromLogs(
+            logs: LogEntry[],
+            data: string[],
+            destinations: string[],
+            txId: BigNumber,
+            values?: BigNumber[],
+        ): void {
+            expect(logs.length).to.eq(data.length + 1);
+            data.forEach((calldata: string, i: number) => {
+                expect(logs[i].address).to.eq(destinations[i]);
+                const contractCallLogArgs = (logs[i] as LogWithDecodedArgs<ContractCallReceiverEventArgs>).args;
+                expect(contractCallLogArgs.functionSelector).to.eq(data[i].slice(0, 10));
+                expect(contractCallLogArgs.data).to.eq(data[i]);
+                const value = values === undefined ? constants.ZERO_AMOUNT : values[i];
+                expect(contractCallLogArgs.value).to.bignumber.eq(value);
             });
-            await assetProxyOwnerWrapper.executeTransactionAsync(erc721AddAuthorizedAddressTxId, owners[0], {
-                gas: constants.MAX_EXECUTE_TRANSACTION_GAS,
+            const executionLog = logs[data.length] as LogWithDecodedArgs<AssetProxyOwnerExecutionEventArgs>;
+            expect(executionLog.event).to.eq('Execution');
+            expect(executionLog.args.transactionId).to.bignumber.eq(txId);
+        }
+        it('should revert if the transaction is not confirmed by the required amount of signers', async () => {
+            const data = [hexRandom()];
+            const destinations = [receiver.address];
+            const results = await assetProxyOwnerWrapper.submitTransactionAsync(data, destinations, signerAddresses[0]);
+            const tx = assetProxyOwner.executeTransaction.awaitTransactionSuccessAsync(results.txId, {
+                from: signerAddresses[1],
```

Comment:
I think it would be good to add a test for the case when the signer that `confirmed` is also the `from` address to cover the bases. Additionally, I think it's a more likely scenario than this one.